Code/Rationals.idr

```diff
@@ -121,8 +132,41 @@ CheckIsQuotientZ (NegS k) (NegS j) x = case ((IsRationalZPOS ((Pos (S k)), (Pos
                                             (Left l) =>  Left (QRproof4 (Pos (S k)) (Pos (S j)) Refl Refl l)
                                             (Right r) => Right ((Pos (S k)), (Pos (S j)))
 
+-- The following section is on simplification of rationals. There was a function written earlier,
+-- called simplifyRational which used the existence of the numeric type 'Double' in Idris so that
+-- simplification could actually be done. This has now been superseded by the functions below as
+-- the GCD with proof has now been implemented.
+
+-- aByd and bByd are taken from the file 'Divisors.idr' (by Shafil)
+
+|||Extracts a/gcd(a,b) from the definition of GCDZ
+aByd:GCDZ a b d ->ZZ
```

Comment:
I think it is better to avoid defining the same function in multiple files.
The problem is that if someone imports rationals.idr and TwoVarDiopheq.idr and uses aByd function, Idris will give an error saying it cannot disambiguate between _ and _

Reply:
Thanks for pointing that out.. I renamed the functions.

Reply:
Thanks, but it would hwve been easier to just import TwoVarDiophEq

Reply:
> Thanks, but it would hwve been easier to just import TwoVarDiophEq

I think it might lead to a cycle in imports (as Rationals is a more basic file and is implemented in many projects, as opposed to TwoVarDiophEq which is more specialized).